File: backend/models/order.go
package models

import "time"

type Order struct {
  ID int `json:"id" gorm:"primary_key"`
  OrderName string `json:"order_name"`
  CustomerID string `json:"customer_id"`
  Customer Customer `json:"customer" gorm:"references:UserID"`
  OrderItems []OrderItem `json:"order_items"`
  CreatedAt time.Time `json:"created_at"`
}

File: frontend/src/models/Order.ts
interface OrderResponse {
  data: Order[]
  limit: number
  page: number
  total_rows: number
  total_pages: number
}

interface Order {
  id: number
  order_name: string
  customer_id: string
  customer: Customer
  order_items: OrderItem[]
  created_at: string
}

interface Customer {
  user_id: string
  login: string
  password: string
  name: string
  company_id: number
  company: Company
  credit_card: string
}
interface Company {
  company_id: number
  company_name: string
}

interface OrderItem {
  id: number
  order_id: number
  price_per_unit: number
  quantity: string
  product: string
  deliveries: Delivery[]
}

interface Delivery {
  id: number
  order_item_id: number
  delivered_quantity: string
}
